cmd/machine-config-daemon/start.go

```diff
@@ -92,8 +97,14 @@ func runStartCmd(cmd *cobra.Command, args []string) {
 		glog.Fatalf("unable to change directory to /: %s", err)
 	}
 
-	stopCh := make(chan struct{})
-	defer close(stopCh)
+	err = daemon.CheckStateOnBoot(stopCh)
+	if err != nil {
+		glog.Fatalf("error checking initial state of node: %v", err)
+	}
+
+	ctx.KubeInformerFactory.Start(ctx.Stop)
```

Comment:
IIUC, there's a race condition here if the node annotation is modified *after* `CheckStateOnBoot()`, but *before* cache is synced and we can start receiving callbacks, right?

Here's an idea: we spawn another goroutine that just listens for notifications on a channel before doing `triggerUpdate()`. Then in `handleUpdate` we send a notification. But crucially, we also send a notification after `WaitForCacheSync()` is done if annotations don't match. That way updates are still always serialized, and we close the race condition.

Reply:
Yes, I thought about this as well, but while testing this it seems that the notifications should still be there for when the informer actually starts listening (Maybe I'm mis-understanding, I used to have a addNode callback as well which triggers despite having happened before the informer starts listening)

Also, during testing, I found that the callback happens quite frequently with the node (once every couple seconds), so updates are generally caught eventually anyways (since we only care about differences in CC vs DC anyways).